packages/wallets/src/evm/wallet.ts

```diff
@@ -26,48 +68,318 @@ export interface ViemWallet {
 }
 
 export class EVMSmartWallet implements ViemWallet {
+    public readonly publicClient: PublicClient<HttpTransport>;
+
     constructor(
-        public readonly publicClient: PublicClient<HttpTransport>,
         public readonly chain: EVMSmartWalletChain,
         private readonly apiClient: ApiClient,
-        private readonly address: Address
-    ) {}
+        private readonly address: Address,
+        private readonly adminSigner: EVMSigner
+    ) {
+        this.publicClient = createPublicClient({
+            chain: toViemChain(chain),
+            transport: http(),
+        });
+    }
 
-    public async balances() {}
-    public async transactions() {}
-    public async nfts() {}
+    public async balances(tokens: Address[]) {
+        return await this.apiClient.getBalance(this.walletLocator, {
+            chains: [this.chain],
+            tokens,
+        });
+    }
+    public async transactions() {
+        const transactions = await this.apiClient.getTransactions(this.walletLocator);
+        return transactions.transactions.filter((transaction) => transaction.walletType === "evm-smart-wallet");
```

Comment:
why do we need this filter?

Reply:
Mostly to simplify the output types.

Reply:
but we're fetching the transactions for that specific walletLocator, so it won't return solana txs as it's not a solana wallet

Reply:
This is about typing. The `apiClient.getTransactions` doesn't discriminate the output type based on the wallet locator type. Could probably add the logic to the `ApiClient`